### PR TITLE
Resizable Editor: Make the editor resizable with arrow keys

### PR DESCRIPTION
**Pull request by t-hamano**
## What?

I found that although the resize handles of the resizable editor are keyboard focusable, pressing the left or right key takes the focus away, making it impossible to resize the editor.

## Why?

This is probably due to the influence of other event handlers.

## How?

As with [here](https://github.com/WordPress/gutenberg/blob/da16125f6f6d7099207fc3d5d2da2ed8c2927022/packages/edit-site/src/components/resizable-frame/index.js#L171), check the key and add `event.preventDefault()`.

## Testing Instructions

- Open the pattern editor.
- Use the tab key to focus the drag handle.
- POresing the left and right key should resize the editor.

## Screenshots or screencast 

https://github.com/user-attachments/assets/aeca7c6e-76bd-49c8-8a18-8a100609c9bd

### CODE REVIEW

**Comment by kevin940726** [comment]
I tested in trunk and the arrow keys are still working correctly 🤔. I must be missing something. Could you share your environment, e.g. the browsers or the setup?

> This is probably due to the influence of other event handlers.

Either way, I think this is still a good improvement!

**Comment by t-hamano** [comment]
Thanks for the review!

> I tested in trunk and the arrow keys are still working correctly 

After further investigation, it appears that, as mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/64529#issuecomment-2291406618), this is due to [the text cursor mode](https://support.google.com/accessibility/answer/10129654?hl=en-GB) in the Chrome browser and Windows OS.

When this mode is enabled, it seems that the focus may move unintentionally using the arrow keys. This mode can be disabled with the `F7` key.

The video below is a test on trunk; in the first half, the text cursor mode is enabled, and in the second half, it is disabled:

https://github.com/user-attachments/assets/f07e50d0-9222-4215-b1a8-a3d3cdaf8770

In any case, it seems that `event.preventDefault();` can prevent the effects of this text cursor mode from being affected.

File: packages/editor/src/components/resizable-editor/resize-handle.js [hunk 1]
```diff
@@ -15,6 +15,11 @@ export default function ResizeHandle( { direction, resizeWidthBy } ) {
 	function handleKeyDown( event ) {
 		const { keyCode } = event;
 
+		if ( keyCode !== LEFT && keyCode !== RIGHT ) {
+			return;
+		}
+		event.preventDefault();
```
Comment (kevin940726) [inline packages/editor/src/components/resizable-editor/resize-handle.js:21]: Should we add a comment referencing this issue, or do you think it's a common practice? I think maybe it's fine to leave it to git blaming to do its work 😅.

**Review by kevin940726 — APPROVED** [review]
Thanks for the update! This change makes sense to me 👍

File: packages/editor/src/components/resizable-editor/resize-handle.js (hunk 1 above)
Comment (t-hamano) [reply packages/editor/src/components/resizable-editor/resize-handle.js:21]: I think if we want to add our own event to an event, it is common to add `event.preventDefault()`, and there are a lot of uncommented `event.preventDefault()` in Gutenberg 😅 Therefore, I personally think that a comment is unnecessary.

**Pull request merged**